framework/src/node/node.ts

```diff
@@ -11,195 +11,188 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
-
-import { Chain, Block } from '@liskhq/lisk-chain';
+import * as path from 'path';
+import { Chain, Block, BlockHeader, BlockAssets } from '@liskhq/lisk-chain';
 import { getNetworkIdentifier } from '@liskhq/lisk-cryptography';
 import { KVStore } from '@liskhq/lisk-db';
-import { Logger } from '../logger';
-import { RegisteredModule, RegisteredSchema } from '../types';
-import { NodeOptions } from './types';
-import { InMemoryChannel } from '../controller/channels';
+import { createLogger, Logger } from '../logger';
 import { Network } from './network';
-import { BaseModule } from '../modules/base_module';
-import { BaseCommand } from '../modules/base_command';
-import { StateMachine } from '../state_machine';
 import { Consensus, CONSENSUS_EVENT_BLOCK_DELETE, CONSENSUS_EVENT_BLOCK_NEW } from './consensus';
-import { BlockGenerateInput, Generator, GenesisBlockGenerateInput } from './generator';
-import { getRegisteredModules, getSchema } from './utils/modules';
+import { BlockGenerateInput, Generator } from './generator';
 import { getEndpointHandlers } from '../endpoint';
-import { ValidatorsAPI, ValidatorsModule } from '../modules/validators';
-import { BFTAPI, BFTModule } from './bft';
+import { BFTModule } from './bft';
 import { APP_EVENT_BLOCK_DELETE, APP_EVENT_BLOCK_NEW } from './events';
-import { NotFoundHandler, RPCServer } from './rpc/rpc_server';
+import { RPCServer, RequestContext } from './rpc/rpc_server';
 import { ChainEndpoint } from './endpoint/chain';
 import { SystemEndpoint } from './endpoint/system';
+import { ABI, InitResponse } from '../abi';
+
+const isEmpty = (value: unknown): boolean => {
+	switch (typeof value) {
+		case 'undefined':
+			return true;
```

Comment:
Should this be false?

Reply:
if the value is undefined, isEmpty should be true

Reply:
seems a bit weird, if we call `isEmpty()` without params it will still return true even though its empty. maybe `null` is better as non-empty value because that will throw error when you call  `isEmpty()`

Reply:
For `isEmpty` function these are expected behavior in my mind, but is it correct?
```
isEmpty() == true
isEmpty('') == true
isEmpty([]) == true
isEmpty({}) == true
```

Reply:
ah true, I mean I was looking something to check in for the emptiness but if there is nothing then also empty so its true, we can resolve this 👍